### PR TITLE
Data Export

### LINKED ISSUE #123: Data export
- We need a feature that can export data in all db's (Users, Software Records, Software Types, Vendor Records).

### PR DESCRIPTION
**Pull request by Uzval-Mallepeddi**
Fixes #123 

- Included options in the dropdown and enabled the export feature only for admin for now.
- Downloads a CSV format of all the data in software_records, vendor_records and software_types

### CODE REVIEW

File: public/software_records.csv [hunk 1]
```diff
@@ -0,0 +1,56 @@
+Software Record,Description,Status,Created on,Software Type,Vendor Record,Departments,Date Implemented,Date of upgrade,Developers,Tech Leads,Product Owners,Languages Used,Production URL,Source Code URL,User Seats,Annual Fees,Support Contract,Hosting Environment,Current Version,Notes,Business Value,IT Quality,Created By
```
Comment (scherztc) [inline public/software_records.csv:1]: @hortongn Should this test data be in public or spec/fixtures?

Comment (hortongn) [reply public/software_records.csv:1]: @scherztc I don't think this is test data.  I think these are .csv files that were generated by the export feature and got checked in to the repository by mistake.  Exporting the files to /public is fine, but we should update .gitignore to ignore any .csv files in that directory.

Comment (Uzval-Mallepeddi) [reply public/software_records.csv:1]: That's a really good catch @scherztc and @hortongn. Those files are being generated by the export feature and I'm saving them locally on the project directory before a user can download them. I'll add them to .gitignore.

File: export_data.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    CSV.open(file, 'w', write_headers: true, headers: headers) do |writer|
+      software_records.each do |software_record|
+        tech_leads = software_record.tech_leads.to_s.gsub!('---', '').to_s
+        tech_leads = tech_leads.gsub!(/\n/, '-').to_s
+
+        departments = software_record.departments.to_s.gsub!('---', '').to_s
+        departments = departments.gsub!(/\n/, '-').to_s
+
+        developers = software_record.developers.to_s.gsub!('---', '').to_s
+        developers = developers.gsub!(/\n/, '-').to_s
```
[the remark this answers is not in the document]
Comment (Uzval-Mallepeddi) [reply export_data.rb:24]: The reason behind that odd formatting is that these fields are multi-valued fields and hence are of the type "Array" in our model. When I tried to convert an array of strings, it resulted in these junk characters. I'll make a workaround on this.

**Comment by scherztc** [comment]
I am okay with merging the PR since I understand the complex problems of exporting nth degree multi valued fields, but at some point during refactor we will need to solve this problem.


Screen Shot 2020-07-14 at 3 25 55 PM

**Comment by Uzval-Mallepeddi** [comment]
@scherztc Makes perfect sense. I agree with the fact that the export features still need some refactoring/data massaging before it gets exported. We'll figure out the scenarios when the data may look weird.

**Pull request merged**